### DISCUSSION TITLE
Expose functionality from Theia Extension to Theia Plugins

### DISCUSSION THREAD

**Discussion by arunswarnam** [eclipse-theia/theia#10620] [Q&A] [answered]
Hi! 
   Can someone suggest what is the best way to expose a method/functionality from Theia IDE to a Theia plugin?  I see [one way](https://github.com/eclipse-theia/theia/discussions/8126) is to expose a command registered by a Theia Extension and let the plugin's call the command by ID but registering a command will make the command show up in the command palette which we don't want to. This is some custom functionality that we want to expose to the plugins and don't want to show as a command to the user.

**Comment by msujew** [comment, marked as answer]
Hi @arunswarnam,

commands don't necessarily show up in the command palette. This only happens when the command has a `label` property. You can create and register commands without defining a label property.

**Reply by arunswarnam** [reply]
Good to know. Thanks @msujew
